### DISCUSSION TITLE
Extreme mode

### DISCUSSION THREAD

**Discussion by LWChris** [octokatherine/word-master#137] [Show and tell]
I always wanted to have a more restrictive mode than HARD, because even in hard mode, you're still allowed to disregard half of the limitations your attempts could impose.

Therefore, I'm proudly introducing: the 🔥🔥🔥 **`EXTREME`** 🔥🔥🔥 mode!

In this mode, gray letters are banned for testing, and yellow letters need to find a new home. 😁

To enable this, I basically rewrote the way the game internally tracks given hints.

This allowed me to another feature I've been wanting for really long: in hard mode, you were required to use all yellow letters and place the green letters in the right place. If you failed to do that, it would just mark your attempt red, like with an unknown word.

More than once I have been dumbfounded thinking "how can that not be a recognized word?", only to realize a few seconds later that the actual problem was a missed positive constraint. I assume this would happen even more often in extreme mode with all the negative constraints; therefore, I changed it so that if the word itself is recognized but doesn't match all hints, rather than highlighting the whole word, it'll highlight the conflicting hints and/or guessed letters.

Please have a go at https://lwchris.github.io/word-master/ and tell me what you think.

@octokatherine I would love to improve this even more; actually, the code has every necessary information to show a tooltip explaining what's wrong when you hover a conflicting hint or letter. I'm just not too familiar with tailwind to easily implement it. Tell me if you're interested in pulling my changes. 😊

**Comment by octokatherine** [comment]
@LWChris that's pretty cool! I'm down to merge in the extreme mode change and highlighting the specific letters in red. 

Yeah having some sort of messaging that says why the word is rejected is totally possible. At one point I did have that in an alert but people were annoyed by it because you had to explicitly dismiss the alert. Just needs some UX work

**Reply by LWChris** [reply]
Cool! I'll add one more commit to create the hints for games that were started with the old code base where hints were not generated after each try, then I'll create a pull request.

**Reply by LWChris** [reply]
See PR #138. I have changed it so that the hints are calculated on demand instead of being stored in the state, because since the storage uses hooks and you may not call hooks from normal code, if or loops, I couldn't figure out how to initialize the values properly. This means it'll work for already started games as well.
